internal/generate/clusterserviceversion/clusterserviceversion_updaters.go

```diff
@@ -315,20 +332,36 @@ func validatingToWebhookDescription(webhook admissionregv1.ValidatingWebhook, de
 	}
 
 	if serviceRef := webhook.ClientConfig.Service; serviceRef != nil {
-		if serviceRef.Port != nil {
-			description.ContainerPort = *serviceRef.Port
+		if serviceRef.Port != nil && ws != nil {
+			for _, port := range ws.Spec.Ports {
+				if *serviceRef.Port == port.Port {
+					description.ContainerPort = port.Port
+					description.TargetPort = &port.TargetPort
+					break
+				}
+			}
 		}
 		description.DeploymentName = depName
 		if description.DeploymentName == "" {
 			description.DeploymentName = strings.TrimSuffix(serviceRef.Name, "-service")
 		}
 		description.WebhookPath = serviceRef.Path
 	}
+
+	// If port values have not been set, then default to 443
+	if description.ContainerPort == 0 && description.TargetPort == nil {
```

Comment:
Modified this so that when `serviceRef` != nil and port in webhook service is not set, we default to 443.